lib/ramble/ramble/cmd/common/info.py

```diff
@@ -6,34 +6,306 @@
 # option. This file may not be copied, modified, or distributed
 # except according to those terms.
 
-from __future__ import print_function
+import fnmatch
 
 import llnl.util.tty.color as color
+from llnl.util.tty.colify import colified
 
+import ramble.util.colors
 import ramble.cmd.common.arguments as arguments
 import ramble.repository
 
-header_color = "@*b"
-plain_format = "@."
+from ramble.util.logger import logger
+
+import enum
+
+supported_formats = enum.Enum("formats", ["text", "lists"])
+
+obj_attribute_map = {
```

Comment:
I think this is totally fine, just a random thought (maybe for future):

Would it be better if these are something provided by the object itself? Can also streamline it a bit for instance by doing auto registration via the directive decorator, etc. Otherwise it might be getting out-of-sync when new things are added on the object side.

Reply:
I actually thought about doing that, but decided against it for the time being. Mostly because some of the attributes (like tags or default_mode) don't really know their actual attribute name via the directive.

This can be seen primarily with the default mode directive, where the attribute we modify is actually `_default_usage_mode` instead of `default_mode`. (Also using `default_mode` as the attribute name breaks the function map of directive names).

I think this is definitely a place that's ripe for cleanup, but I'm not sure what the best way to do this is currently.

Reply:
Makes sense, I guess one possible approach is to enhance the directive to allow passing in the attr name (a bit overlap with the current `dict` argument.) But yeah the current way is totally fine.